js/wordfinder.js

```diff
@@ -0,0 +1,113 @@
+let answer; // this is a global variable, the result of the pickWord() function. This is bad practice.
+let usedWords = [];
+let allData = {};
+const submitButton = document.querySelector('.submit-btn');
+const results = document.querySelector('#results');
+const correct = document.createTextNode(`Correct!`);
+const incorrect = document.createTextNode(`Incorrect!`);
+
+const startButton = document.querySelector('#start');
+
+const handleButtonClick = (event) => {
+
+    const instructions = document.querySelector('.instructions');
+    const game = document.querySelector('.game');
+
+    if (instructions.style.display === 'none') {
+        instructions.style.display = 'block';
+        game.style.display = 'none';
+        startButton.innerText = 'hide';
+    }
+    else {
+        instructions.style.display = 'none';
+        game.style.display = 'block';
+    }
+}
+
+startButton.addEventListener('click', handleButtonClick);
+
+// Fetch data from json file and return the words object without the description.
+const extractWords = (response) => {
+    const wordList = response.json();
+    return wordList;
+}
+
+const processWords = (wordList) => {
+    allData = wordList.data;
+    console.log(allData);
+    return allData;
+}
+
+fetch('json/words.json')
+    .then(extractWords)
+    .then(processWords);
+
+const playButton = document.querySelector('#start');
+
+const handlePlayButtonClick = (event) => {
+    event.preventDefault();
+    playAgain.style.display ='none';
+    answer = pickWord(allData);
+    console.log(answer);
+    return answer;
+}
+
+playButton.addEventListener('click', handlePlayButtonClick);
+
+const playAgain = document.querySelector('#again');
+playAgain.style.display ='none';
+
+playAgain.addEventListener('click', handlePlayButtonClick);
+
+function pickWord(allData) {
+    let allKeys = (Object.keys(allData));
+    let randomWord = allKeys[Math.floor(Math.random() * allKeys.length)];
+    if (usedWords.length === allData.length) {
```

Comment:
This if statement seems unnecessary. Or it should indicate something to the user via the page rather than the console.